crates/hyperqueue/src/dashboard/ui/screen/controller.rs

```diff
@@ -0,0 +1,37 @@
+use crate::dashboard::events::{DashboardEvent, ScreenChangeEvt};
+use tako::WorkerId;
+use tokio::sync::mpsc::UnboundedSender;
+
+#[derive(Clone, Copy)]
+pub enum DashboardScreenState {
+    ClusterOverviewScreen,
+    WorkerOverviewScreen(WorkerId),
+}
+
+pub struct ScreenController {
+    event_channel: UnboundedSender<DashboardEvent>,
+}
+
+impl ScreenController {
+    pub fn new(event_channel: UnboundedSender<DashboardEvent>) -> Self {
+        Self { event_channel }
+    }
+
+    pub fn show_worker_screen(&mut self, worker_id: WorkerId) {
+        self.send_change_event(ScreenChangeEvt::create(
+            DashboardScreenState::WorkerOverviewScreen(worker_id),
+        ))
+    }
+
+    pub fn show_cluster_overview(&mut self) {
+        self.send_change_event(ScreenChangeEvt::create(
+            DashboardScreenState::ClusterOverviewScreen,
+        ))
+    }
+
+    fn send_change_event(&mut self, event: ScreenChangeEvt) {
+        if let Err(err) = self.event_channel.send(DashboardEvent::ScreenChange(event)) {
```

Comment:
Let's use `expect` and crash the app if the send fails. It's a non-recoverable situation.